kapitan/refs/secrets/azkms.py

```diff
@@ -0,0 +1,179 @@
+"azkms secret module"
+
+import os
+import logging
+import base64
+
+from azure.keyvault.keys.crypto import CryptographyClient, EncryptionAlgorithm
+from azure.keyvault.keys import KeyClient
+from azure.identity import DefaultAzureCredential
+
+from kapitan.refs.base64 import Base64Ref, Base64RefBackend
+from kapitan.refs.base import RefError
+from kapitan import cached
+from kapitan.errors import KapitanError
+
+logger = logging.getLogger(__name__)
+
+
+class AzureKMSError(KapitanError):
+    """
+    Generic Azure Key Vault error
+    """
+
+    pass
+
+
+def azkms_obj(key_id):
+    """
+    Return Azure Key Vault Object
+    """
+    # e.g of key_id https://kapitanbackend.vault.azure.net/keys/myKey/deadbeef
+    if not cached.azkms_obj:
+
+        attrs = key_id.split("/")
```

Comment:
Wouldn't it be cleaner to parse this with `urllib.parse.urlparse()`?

Reply:
Even even with `urllib.parse.urlparse()` we'll have to `slpit()` to get the `key_name` and `key_version`. Plus an added dependency of `urllib`. Besides this, the URL must be introduced by '//' for `urllib` to correctly parse it.

Reply:
Yes, we'd still need split, but urllib is part of the standard library and this works just fine:

```python
In [2]: urllib.parse.urlparse("https://kapitanbackend.vault.azure.net/keys/myKey/deadbeef").path.split("/")
Out[2]: ['', 'keys', 'myKey', 'deadbeef']

In [3]: urllib.parse.urlparse("kapitanbackend.vault.azure.net/keys/myKey/deadbeef").path.split("/")
Out[3]: ['kapitanbackend.vault.azure.net', 'keys', 'myKey', 'deadbeef']
```